README.md

```diff
@@ -112,20 +115,77 @@ public void Configure(IApplicationBuilder app)
+    app.UseGraphQLGraphiQL();
 
-    // use graphql-playground middleware at default path /ui/playground
+    // use GraphQL Playground middleware at default path /ui/playground with default options
     app.UseGraphQLPlayground();
 
-    // use altair middleware at default path /ui/altair
+    // use Altair middleware at default path /ui/altair with default options
     app.UseGraphQLAltair();
     
-    // use voyager middleware at default path /ui/voyager
+    // use Voyager middleware at default path /ui/voyager with default options
     app.UseGraphQLVoyager();
 }
 ```
 
+2. With routing:
+
+```c#
+public void ConfigureServices(IServiceCollection services)
+{
+    // Add GraphQL services and configure options
+    services
+        .AddRouting()
+        .AddSingleton<IChat, Chat>()
+        .AddSingleton<ChatSchema>()
+        .AddGraphQL((options, provider) =>
+        {
+            options.EnableMetrics = Environment.IsDevelopment();
+            var logger = provider.GetRequiredService<ILogger<Startup>>();
+            options.UnhandledExceptionDelegate = ctx => logger.LogError("{Error} occurred", ctx.OriginalException.Message);
+        })
+        // It is required when both GraphQL HTTP and GraphQL WebSockets middlewares are mapped to the same endpoint (by default 'graphql').
+        .AddDefaultEndpointSelectorPolicy()
+        // Add required services for GraphQL request/response de/serialization
+        .AddSystemTextJson() // For .NET Core 3+
+        .AddNewtonsoftJson() // For everything else
+        .AddErrorInfoProvider(opt => opt.ExposeExceptionStackTrace = Environment.IsDevelopment())
+        .AddWebSockets() // Add required services for web socket support
+        .AddDataLoader() // Add required services for DataLoader support
+        .AddGraphTypes(typeof(ChatSchema)); // Add all IGraphType implementors in assembly which ChatSchema exists 
+}
+
+public void Configure(IApplicationBuilder app)
+{
+    // this is required for websockets support
+    app.UseWebSockets();
+
+    // this is required for ASP.NET Core routing
+    app.UseRouting();
+            
+    app.UseEndpoints(endpoints =>
+    {
+        // map websocket middleware for ChatSchema at default path /graphql
+        endpoints.MapGraphQLWebSockets<ChatSchema>();
+
+        // map HTTP middleware for ChatSchema at default path /graphql
+        endpoints.MapGraphQL<ChatSchema, GraphQLHttpMiddlewareWithLogs<ChatSchema>>();
+
+        // map GraphQL Playground middleware at default path /ui/playground with default options
+        endpoints.MapGraphQLPlayground();
+
+        // map GraphiQL middleware at default path /ui/graphiql with default options
+        endpoints.MapGraphQLGraphiQL();
+
+        // map Altair middleware at default path /ui/altair with default options
+        endpoints.MapGraphQLAltair();
+
+        // map Voyager middleware at default path /ui/voyager with default options
+        endpoints.MapGraphQLVoyager();
+}
+```
```

Comment:
Worth taking a look at @SimonCropp 's MarkdownSnippet for these. https://github.com/SimonCropp/MarkdownSnippets

Reply:
I roughly understood what this is for, but it seems too complicated to me. It is not difficult for me to update the readme manually.